doc/source/qip-vqa.rst

```diff
@@ -0,0 +1,119 @@
+
+.. image:: /figures/vqa_circuit_with_x.png
+   
+
+--------------
+
+
+Optimisation Loop
+=================
+
+After specifying a cost method and function to the :class:`.VQA` instance, there are various options for optimization of the free circuit parameters. Calling :meth:`.VQA.optimize_parameters` will begin the optimization process and return an :class:`.OptimizationResult` instance. By default, the method will randomize initial parameters, and use the non-gradient-based ``COBYLA`` method for parameter optimization. Users can specify:
+
+
+  * **Initial parameters**. Given as a list, with length corresponding to the number of free parameters in the circuit. The number of free parameters can be computed automatically with the :meth:`.VQA.get_free_parameters` method. Alternatively, the string 'zeros' will initialize all parameters as 0; and 'random' will initialize parameters randomly between 0 and 1. Defaults to 'random'.
+
+  * **Optimization method**. This can be a string referring to a pre-defined ``SciPy`` method `listed here <https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html>`_, or a callable function.
+
+  * **Jacobian computation**. A flag will tell the optimization method to compute the Jacobian at each step, which is passed to the optimizer so that it can use gradient information.
+
+  * **Layer-by-layer training**. Optimize parameters for the circuit with only a single layer, and hold these fixed while adding additional layers, up to ``VQA.n_layers``.
+
+  * **Bounds and constraints**. To be passed to the optimizer.
+
+The :class:`.OptimizationResult` class provides information about the completed optimization process. For example, the probability amplitudes of different measurement outcomes of the circuit post-optimization can be plotted with :meth:`.OptimizationResult.plot`.
+
+Below, we run an optimization on a toy circuit, tuning a parameterized :math:`x`-rotation gate to try to maximise the probability amplitude of the :math:`|1\rangle` state.
+
+.. testcode::
+
+  from qutip_qip.vqa import VQA, VQABlock
+  from qutip import sigmax, sigmaz
+
+  circ = VQA(n_qubits=1, cost_method="OBSERVABLE")
+  # Our cost function will be  <psi(t)| sigma_z |psi(t)>
+  circ.cost_observable = sigmaz()
+
+  # Our circuit is just an x-rotation gate: e^{-i * t/2 * X}
+  circ.add_block(
+          VQABlock(sigmax() / 2)
+          )
+
+  # Optimize with the SciPy in-built BFGS method
+  result = circ.optimize_parameters(
+          method="BFGS", use_jac=True
+          )
+
+  # Access the SciPy results to output the parameter found
+  print(f'Parameter found: {round(result.res.x[0], 2)}')
+  result.plot()
+
+**Output**:
+
+.. testoutput::
+  :options: +NORMALIZE_WHITESPACE
+
+  Parameter found: 3.14
+
+.. image:: /figures/vqa_circuit_optimization_output.png
```

Comment:
You don't have to import his figure. Using the plot derivative you can directly show the matplotlib figure you generated from the code (https://qutip-qip.readthedocs.io/en/latest/contribution-docs.html#id4). Sorry if my previous comment misleads you. I meant only the circuit figure.

Reply:
Got it, thanks! I've switched to using the plot directive for this section now.